include/swift/Reflection/ReflectionContext.h

```diff
@@ -520,15 +533,14 @@ class ReflectionContext
     if (MagicBytes[0] == 'M' && MagicBytes[1] == 'Z') {
       return readPECOFF(ImageStart);
     }
-    
+
     // ELF.
-    if (MagicBytes[0] == llvm::ELF::ElfMagic[0]
-        && MagicBytes[1] == llvm::ELF::ElfMagic[1]
-        && MagicBytes[2] == llvm::ELF::ElfMagic[2]
-        && MagicBytes[3] == llvm::ELF::ElfMagic[3]) {
-      return readELF(ImageStart);
-    }
-    
+    assert(MagicBytes[0] == llvm::ELF::ElfMagic[0] &&
```

Comment:
Is this ok in `addImage`? Since we now require the file buffer in `readELF`, we can't really can't really call it from here anymore.

Reply:
I don't fully understand the original question — can you elaborate?

Separate from that, we shouldn't crash/assert on invalid input, but return an error instead.

Reply:
I mean that `addImage` used to call `readELF`, but since we require the file buffer now this can't really be done anymore, and was wondering if that's ok. 

I removed the call and inserted the assert, I can just return false instead.